### PR TITLE
Remove silent mode

### PR DESCRIPTION
**Pull request by notlee**
Remove silent mode for the primary mixin.

Origami spec v2 components will require a primary mixin and
will no longer recommend the silent mode variable.

Related proposal: https://github.com/Financial-Times/origami/pull/121
Related spec suggestion: https://github.com/Financial-Times/origami-website/pull/273#discussion_r626746297

_See commit messages for more re. the npm install change_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ Runs JavaScript and Sass tests.
 
 * If `--debug` is set, the test runner will not exit automatically to allow debugging of the tests.
 
-Checks Sass supports [silent and non-silent compilation modes](https://origami.ft.com/spec/v1/sass/#sass-silent-mode).
+Checks Sass supports [includes a primary mixin](https://origami.ft.com/spec/v2/components/sass/#primary-mixin).
```
Comment (JakeChampion) [inline README.md:109]:
```suggestion
Checks Sass [includes a primary mixin](https://origami.ft.com/spec/v2/components/sass/#primary-mixin).
```

File: test/integration/test/test.test.js [hunk 2]
```diff
@@ -7,36 +7,90 @@ const path = require('path');
...
 const rimraf = require('../helpers/delete');
 const obtBinPath = require('../helpers/obtpath');
+const fs = require('fs');
+const { promisify } = require('util');
+const mkdtemp = promisify(fs.mkdtemp);
+const os = require('os');
+const proclaim = require('proclaim');
 
-describe('obt test', function () {
+describe.only('obt test', function () {
```
Comment (notlee) [inline test/integration/test/test.test.js:16]:
```suggestion
describe('obt test', function () {
```

File: lib/tasks/test-sass-compilation.js [hunk 3]
```diff
@@ -5,42 +5,72 @@ const buildSass = require('./build-sass');
...
-		sassIncludePaths: [cwd],
-		brand: brand,
-		buildFolder: 'disabled',
-		cwd: cwd,
-		sourcemaps: false,
-		outputStyle: 'compressed' // to remove sass multiline/loud comments from the css output
-	};
+	const primaryMixinName = camelCase(name);
+	const primaryMixinErrorCode = `OBT_NO_PRIMARY-MIXIN-${primaryMixinName}`;
+	const sassContent = fs.readFileSync(mainSassFilePath, 'utf8');
```
Comment (JakeChampion) [inline lib/tasks/test-sass-compilation.js:25]: `obt test` runs all the test sub-tasks at the same time, having this use a synchronous file read will cause the other sub-tasks to not run whilst this file is being read. Can we make this use an async version instead, that way other tasks/sub-tasks can run whilst the sass is being tested.

We would need to require the promisified version of `fs` - `require('fs').promise` and then use it like `await fs.readFile`

File: lib/tasks/test-sass-compilation.js [hunk 4]
```diff
@@ -5,42 +5,72 @@ const buildSass = require('./build-sass');
...
+				@error '${primaryMixinErrorCode}';
+			};
+			@include ${primaryMixinName}();
+		` : ''}
+	`;
+
+	// Build Sass
 	try {
-		const css = await buildSass(sassConfig);
+		const result = execa.sync(ftSass, [
```
Comment (JakeChampion) [inline lib/tasks/test-sass-compilation.js:42]: Same thing here, could we change this from a sync call to an async call as to not block the event-loop as other test sub-tasks could be running at this time as well

File: test/integration/test/test.test.js (hunk 2 above)
Comment (JakeChampion) [inline test/integration/test/test.test.js]:
```suggestion
describe('obt test', function () {
```

**Comment by notlee** [comment]
Sorry this is a chunky PR. Test changes outside silent mode have already been reviewed here:
https://github.com/Financial-Times/origami-build-tools/pull/1020

File: lib/tasks/test-sass-compilation.js [hunk 5]
```diff
@@ -62,26 +91,24 @@ module.exports = function (cfg) {
 				{
 					title: 'Tests starting...',
 					task: async (context, task) => {
-						const brands = await files.getModuleBrands();
-						task.title = `Testing compilation with silent mode on`;
+						let brands = await files.getModuleBrands();
+						brands = brands.length > 0 ? brands : [
```
Comment (notlee) [inline lib/tasks/test-sass-compilation.js:95]: Let's leave updating brands to meet new spec changes, e.g. 'universal', as a separate PR

**Review by JakeChampion — APPROVED** [review]

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v11.0.0-beta.31 :tada:

**Pull request merged**